x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -623,16 +623,20 @@ export const LensTopNavMenu = ({
             share.toggleShareContextMenu({
               anchorElement,
               allowEmbed: false,
-              allowShortUrl: false, // we'll manage this implicitly via the new service
-              shareableUrl: shareableUrl || '',
-              shareableUrlForSavedObject: savedObjectURL.href,
+              allowShortUrl: false,
+              delegatedShareUrlHandler: () => {
+                return shareUrlEnabled ? shareableUrl! : savedObjectURL.href;
```

Comment:
I assume that if `shareUrlEnabled` is false the share feature would not be available at all.

I think the correct code here is this one:
```suggestion
                return isCurrentStateDirty ? shareableUrl! : savedObjectURL.href;
```